### ISSUE TITLE
Multiple playlist example

### ISSUE THREAD

**Issue by moonwave99** [andrewrk/node-groove#17]
I am developing an [electron](https://github.com/atom/electron) based app, that delegates all audio control to node-groove.

I have multiple tabs, each holding a different `GroovePlaylist`:

![playa.jpg](https://www.dropbox.com/s/v20ewhww5faklbg/playa%20%281%29.jpg?dl=0&raw=1)

My control flow is:

- create a `GroovePlayer`;
- create a `GroovePlaylist`;
- load some file into playlist;
- when I double click on some entry of such playlist, or on the play button -> attach playlist to the player;
- when I double click on **another playlist** entry -> detach player -> attach the new playlist.

I can see correct playback information on top, but I get no audio output, nor playback progress information.

Relevant parts of the code:

`Player.js`:

    attach(){
      return new Promise((resolve, reject)=>{
        if(this.attached){
          resolve()
        }else if(!this.playlist){
          reject(new Error('No playlist set!'))
        }else{
          this.player.attach(this.playlist, (err)=>{
            if(err){
              reject(err)
            }else{
              this.attached = true
              resolve()
            }
          })
        }    
      })
    }
    detach(){
      return new Promise((resolve, reject)=>{
        if(!this.attached){
          resolve()
        }else if(!this.playlist){
          reject(new Error('No playlist to detach!'))
        }else{
          this.player.detach((err)=>{
            if(err){
              reject(err)
            }else{
              this.attached = false
              resolve()
            }
          })
        }
      })
    } 
    play() {
      this.attach().then(()=>{
        this.playlist.play()
      })
    }
    goto(id) {
      var item = _.findWhere(this.playlist.items(), { id: id })
      if(item){
        this.playlist.seek(item, -1)
        !this.playlist.playing() && this.play()
      }
    }

`PlaylistStore.js`:

    // this is the PLAY_FILE action - I detach player, then I seek to the desired item
    if(action.playlist.groovePlaylist !== _player.playlist){
      _player.detach().then(()=>{
        _player.playlist = action.playlist.groovePlaylist
        _player.goto(action.id)
      }).catch((error)=>{
    
      })
    }else{
      _player.goto(action.id)
    }
    
---

Did I get anything wrong, with how do playlists work? Should I instead have just a _single_ playlist, from whom constantly inserting and removing?

**Comment by andrewrk** [comment]
Yes you should have a single playlist and do inserting and removing. Does the playlist example js file work in node-groove?

**Comment by moonwave99** [comment]
Yes it does ^^

At the moment, I changed it to work with a single groovePlaylist, that I clear and refill every time I switch context - works like a charm so far.  
Perhaps it wouldn't harm to add a comment in the README about the management of multiple playlists.

Btw I really appreciate your work, I can finally work on a reliable player that suits my needs, after many years!

**Comment by andrewrk** [comment]
Here is how Groove Basin manages the groove playlist: https://github.com/andrewrk/groovebasin/blob/84d77ce10ea9e7a50b304bfb6cabd23acde7e493/lib/player.js#L2439

It's rather complicated, but I think it's as simple as it can be while still being correct and avoiding gaps/glitches in playback at edge cases.

That said, I think I encourage you to implement your own code and see where it goes.

Fair point about the README, I'll do that.
